### PR TITLE
get-categories completed

### PR DESCRIPTION
**Pull request by isamotta**

### CODE REVIEW

File: controllers/categories.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const { fetchAllCategories } = require('../models/categories')
+
+const getAllCategories = (req, res, next) => {
+    fetchAllCategories().then((result) => {
+        res.status(200).send({ categories: result })
+    })
+}
```
Comment (NRMSMITH) [inline controllers/categories.js:9]: How does your controller _catch_ your errors?

File: app.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const express = require('express');
+const { getAllCategories } = require('./controllers/categories')
+const app = express();
+
+app.get('/api/categories', getAllCategories)
+
```
Comment (NRMSMITH) [inline app.js:18]: Error handling middleware can be included here! It would be appropriate to include a 500 error handler here.

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
Request changes: error handling and catching errors :)

File: controllers/categories.js (hunk 1 above)
Comment (isamotta) [reply controllers/categories.js:9]: Added catch block

File: app.js (hunk 2 above)
Comment (isamotta) [reply app.js:18]: Added middleware function to handle 500 error.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    return db.end()
+})
+
+describe("/not-a-route", () => {
+    test('responds with a 404 status code when using an API that does not exist', () => {
+        return request(app)
+            .get('/not-a-route')
+            .expect(404);
+    });
+})
```
Comment (NRMSMITH) [inline __tests__/app.test.js:24]: You should be expecting a message saying 'does not exist' here :)  Also 'API' should be 'endpoint'.

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
-> Please ensure that your test is checking that you are getting back a message.

File: __tests__/app.test.js (hunk 3 above)
Comment (isamotta) [reply __tests__/app.test.js:24]: added a message

**Review by EddTheDuck — APPROVED** [review]
Nice job :)

**Pull request merged**